src/Datasource/ConnectionInterface.php

```diff
@@ -17,6 +17,13 @@
 /**
  * This interface defines the methods you can depend on in
  * a connection.
+ *
+ * The following methods will be added in version 4.0.0
+ *
+ * @method $this setDriver($driver, $config = []);
+ * @method \Cake\Database\Driver getDriver();
```

Comment:
I don't think the driver methods belong to the ConnectionInterface. They are only really used by the PDO/database connection.

https://github.com/FriendsOfCake/process-mq/blob/master/src/Connection/RabbitMQConnection.php for example would not use a get/setDriver method, but could (should) implement the interface.